libs/application/templates/driving-learners-permit/src/fields/LookupStudent/index.tsx

```diff
@@ -0,0 +1,218 @@
+        )
+      }
+    },
+  })
+
+  const studentIsMentorable = getValues(fieldNames.studentIsMentorable)
+
+  // Clear inital errors on mount
+  useEffect(() => {
+    clearErrors()
+  }, [])
+
+  const studentNationalId: string = watch(fieldNames.studentNationalId)
+  const studentName: string = watch(fieldNames.studentName)
+
+  useEffect(() => {
+    if (studentNationalId?.length === 10) {
+      const isValidSSN = kennitala.isPerson(studentNationalId)
+      if (isValidSSN) {
+        if (fakeData?.useFakeData === YES) {
+          if (
+            fakeData?.mentorableStudents
+              ?.split(',')
+              .map((v) => v.trim())
+              .includes(studentNationalId)
+          ) {
+            setValue(fieldNames.studentIsMentorable, 'isMentorable')
+            setValue(fieldNames.studentName, 'Æfinga Leyfisbur')
+          } else {
+            setValue(fieldNames.studentIsMentorable, 'isNotMentorable')
+            setValue(fieldNames.studentName, 'Óleyf Keyra Vagnsdóttir')
+          }
+        } else {
+          const identityInput = {
+            variables: {
+              input: {
+                nationalId: studentNationalId,
+              },
+            },
+          }
+          const studentLookupInput = {
+            variables: {
+              input: {
+                studentSSN: studentNationalId,
+              },
+            },
+          }
+          getIdentity(identityInput)
+          getStudentMentorability(studentLookupInput)
+          setValue(fieldNames.studentIsMentorable, 'loading')
+        }
+      } else if (studentName !== '') {
+        setValue(fieldNames.studentName, '')
+      }
+    } else if (studentName !== '') {
+      setValue(fieldNames.studentName, '')
+    }
+  }, [studentName, studentNationalId, getIdentity, setValue])
+
+  return (
+    <Box>
+      <GridRow marginBottom={2} marginTop={2}>
+        <GridColumn span="6/12">
+          <InputController
+            id={fieldNames.studentNationalId}
+            name={fieldNames.studentNationalId}
+            label={formatMessage(m.studentNationalIdLabel)}
+            format="######-####"
+            defaultValue=""
+            backgroundColor="blue"
+            loading={identityQueryLoading}
+          />
+        </GridColumn>
+        <GridColumn span="6/12">
+          <InputController
+            id={fieldNames.studentName}
+            name={fieldNames.studentName}
+            label={formatMessage(m.studentNameLabel)}
+            readOnly
+            defaultValue=""
+          />
+        </GridColumn>
+        <Box hidden={true}>
+          <InputController
+            id={fieldNames.studentIsMentorable}
+            name={fieldNames.studentIsMentorable}
+            readOnly
+            defaultValue="default"
+          />
+        </Box>
+      </GridRow>
+
+      <GridRow marginTop={5}>
+        <GridColumn span="12/12">
+          {studentIsMentorable !== 'default' && (
```

Comment:
The variable name suggest a boolean value, should this be renamed to something else?

Reply:
yes, probably